code/modules/power/engines/singularity/singularity.dm

```diff
@@ -336,7 +336,7 @@
 
 	var/movement_dir = pick(GLOB.alldirs - last_failed_movement)
 
-	if(get_turf(src) == target || !target)
+	if(!target || get_dist(src, target) <= 2 || get_turf(src) == target)
```

Comment:
Isn't `get_turf(src) == target` redundant if we're already checking for `get_dist(src, target) <= 2`?

Reply:
Yep, should be able to be just handled by the range check

Reply:
Derp, you're right. Fixed.